### ISSUE TITLE
Firefox can't navigate with proxy

### ISSUE THREAD

**Issue by Sue9445** [microsoft/playwright#917]
Hi,

I tried to launch firefox with proxy prefences and I set the TLS version as 1 to avoid SSL errors. (I know Firefox will drop support for TLS 1.0)

When request interception enabled, requests cannot be handled.

When i tried to load page without request interception, I am getting Firefox Network Error html output. (https://pastebin.com/8cNMve8v)

My proxy works very well with Chromium.

**Prefs:**
```
user_pref("security.tls.version.min", 1);
user_pref("network.stricttransportsecurity.preloadlist", false);
user_pref("network.proxy.type", 1);
user_pref("network.proxy.share_proxy_settings", true);
user_pref("network.proxy.http", "127.0.0.1");
user_pref("network.proxy.http_port", 39242);
user_pref("network.proxy.ssl", "127.0.0.1");
user_pref("network.proxy.ssl_port", 39242);
```

**Comment by pavelfeldman** [comment]
Could you clarify how you configure Firefox to use your proxy settings? Also, note the `ignoreHTTPSErrors` param in [`browser.newContext`](https://github.com/microsoft/playwright/blob/v0.10.0/docs/api.md#browsernewcontextoptions).

**Comment by Sue9445** [comment]
I also used the ignoreHTTPSErrors parameter but I found my mistake, firefox doesn't support networkidle0 yet.

Thank you.

**Comment by Sue9445** [comment]
When i try again with request interception, requests still cannot be handled.

Here is my full code:

 ```
(async () => {

	const pw = require("playwright");
	const os = require("os");
	const path = require("path");
	const fs = require("fs");

	const userDataDir = `${os.tmpdir()}/pw-firefox-test/'}`;

	if(!fs.existsSync(userDataDir)){
		fs.mkdirSync(userDataDir, { recursive: true });
	}

	let prefs = `
	  	user_pref("security.tls.version.min", 1);
		user_pref("network.stricttransportsecurity.preloadlist", false);
		user_pref("network.proxy.type", 1);
		user_pref("network.proxy.share_proxy_settings", true);
		user_pref("network.proxy.http", "127.0.0.1");
		user_pref("network.proxy.http_port", 30921);
		user_pref("network.proxy.ssl", "127.0.0.1");
		user_pref("network.proxy.ssl_port", 30921);
	`;

	fs.writeFileSync(path.join(userDataDir, "./pref.js"), prefs);
	fs.writeFileSync(path.join(userDataDir, "./user.js"), prefs);

	const browser = await pw.firefox.launch({
		headless: true,
		userDataDir: userDataDir,
	});

	const context = await browser.newContext({
		ignoreHTTPSErrors: true,
		bypassCSP: true,
		userAgent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:72.0) Gecko/20100101 Firefox/72.0",
		viewport: {
			width: 1920,
			height: 1040,
		}
	});

	const page = await context.newPage();

	await page.setRequestInterception(true);

	page.on("request", request => {
		let type = request.resourceType();

		if(type == "image" || type == "font"){
			request.abort();
		} else {
			request.continue();
		}
	});

	await page.goto("https://www.google.com");

	await page.waitFor(5000);

	await page.screenshot({path: "test.png", fullPage: true});

	await browser.close();
})();
 ```

**Comment by pavelfeldman** [comment]
We overwrite your prefs as per https://github.com/microsoft/playwright/issues/857. We'll fix it.

**Comment by Sue9445** [comment]
It's not working with playwright default prefs too.

```javascript
(async () => {
	const pw = require("playwright");
	const os = require("os");
	const path = require("path");
	const fs = require("fs");

	const userDataDir = `${os.tmpdir()}/pw-firefox-test/'}`;

	if(!fs.existsSync(userDataDir)){
		fs.mkdirSync(userDataDir, { recursive: true });
	}

	let prefs = `
		user_pref("app.normandy.api_url", "");
		user_pref("app.update.checkInstallTime", false);
		user_pref("app.update.disabledForTesting", true);
		user_pref("apz.content_response_timeout", 60000);
		user_pref("browser.contentblocking.features.standard", "-tp,tpPrivate,cookieBehavior0,-cm,-fp");
		user_pref("browser.dom.window.dump.enabled", true);
		user_pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
		user_pref("browser.newtabpage.enabled", false);
		user_pref("browser.pagethumbnails.capturing_disabled", true);
		user_pref("browser.safebrowsing.blockedURIs.enabled", false);
		user_pref("browser.safebrowsing.downloads.enabled", false);
		user_pref("browser.safebrowsing.malware.enabled", false);
		user_pref("browser.safebrowsing.passwords.enabled", false);
		user_pref("browser.safebrowsing.phishing.enabled", false);
		user_pref("browser.search.update", false);
		user_pref("browser.sessionstore.resume_from_crash", false);
		user_pref("browser.shell.checkDefaultBrowser", false);
		user_pref("browser.startup.homepage", "about:blank");
		user_pref("browser.startup.homepage_override.mstone", "ignore");
		user_pref("browser.startup.page", 0);
		user_pref("browser.tabs.disableBackgroundZombification", false);
		user_pref("browser.tabs.warnOnCloseOtherTabs", false);
		user_pref("browser.tabs.warnOnOpen", false);
		user_pref("browser.uitour.enabled", false);
		user_pref("browser.urlbar.suggest.searches", false);
		user_pref("browser.usedOnWindows10.introURL", "");
		user_pref("browser.warnOnQuit", false);
		user_pref("datareporting.healthreport.about.reportUrl", "http://dummy.test/dummy/abouthealthreport/");
		user_pref("datareporting.healthreport.documentServerURI", "http://dummy.test/dummy/healthreport/");
		user_pref("datareporting.healthreport.logging.consoleEnabled", false);
		user_pref("datareporting.healthreport.service.enabled", false);
		user_pref("datareporting.healthreport.service.firstRun", false);
		user_pref("datareporting.healthreport.uploadEnabled", false);
		user_pref("datareporting.policy.dataSubmissionEnabled", false);
		user_pref("datareporting.policy.dataSubmissionPolicyAccepted", false);
		user_pref("datareporting.policy.dataSubmissionPolicyBypassNotification", true);
		user_pref("devtools.jsonview.enabled", false);
		user_pref("dom.disable_open_during_load", false);
		user_pref("dom.file.createInChild", true);
		user_pref("dom.ipc.reportProcessHangs", false);
		user_pref("dom.max_chrome_script_run_time", 0);
		user_pref("dom.max_script_run_time", 0);
		user_pref("extensions.autoDisableScopes", 0);
		user_pref("extensions.enabledScopes", 5);
		user_pref("extensions.getAddons.cache.enabled", false);
		user_pref("extensions.installDistroAddons", false);
		user_pref("extensions.screenshots.disabled", true);
		user_pref("extensions.update.enabled", false);
		user_pref("extensions.update.notifyUser", false);
		user_pref("extensions.webservice.discoverURL", "http://dummy.test/dummy/discoveryURL");
		user_pref("focusmanager.testmode", true);
		user_pref("general.useragent.updates.enabled", false);
		user_pref("geo.provider.testing", true);
		user_pref("geo.wifi.scan", false);
		user_pref("hangmonitor.timeout", 0);
		user_pref("javascript.options.showInConsole", true);
		user_pref("gfx.color_management.mode", 0);
		user_pref("gfx.color_management.rendering_intent", 3);
		user_pref("media.gmp-manager.updateEnabled", false);
		user_pref("network.cookie.cookieBehavior", 0);
		user_pref("network.http.prompt-temp-redirect", false);
		user_pref("network.http.speculative-parallel-limit", 0);
		user_pref("network.manage-offline-status", false);
		user_pref("network.sntp.pools", "dummy.test");
		user_pref("plugin.state.flash", 0);
		user_pref("privacy.trackingprotection.enabled", false);
		user_pref("remote.enabled", true);
		user_pref("security.certerrors.mitm.priming.enabled", false);
		user_pref("security.fileuri.strict_origin_policy", false);
		user_pref("security.notification_enable_delay", 0);
		user_pref("services.settings.server", "http://dummy.test/dummy/blocklist/");
		user_pref("signon.autofillForms", false);
		user_pref("signon.rememberSignons", false);
		user_pref("startup.homepage_welcome_url", "about:blank");
		user_pref("startup.homepage_welcome_url.additional", "");
		user_pref("toolkit.cosmeticAnimations.enabled", false);
		user_pref("toolkit.telemetry.server", "https://dummy.test/dummy/telemetry/");
		user_pref("toolkit.startup.max_resumed_crashes", -1);
	  	user_pref("security.tls.version.min", 1);
		user_pref("network.stricttransportsecurity.preloadlist", false);
		user_pref("network.proxy.type", 1);
		user_pref("network.proxy.share_proxy_settings", true);
		user_pref("network.proxy.http", "127.0.0.1");
		user_pref("network.proxy.http_port", 36535);
		user_pref("network.proxy.ssl", "127.0.0.1");
		user_pref("network.proxy.ssl_port", 36535);
	`;

	fs.writeFileSync(path.join(userDataDir, "./user.js"), prefs);

	const browser = await pw.firefox.launch({
		headless: true,
		userDataDir: userDataDir,
	});

	const context = await browser.newContext({
		ignoreHTTPSErrors: true,
		bypassCSP: true,
		userAgent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:72.0) Gecko/20100101 Firefox/72.0",
		viewport: {
			width: 1920,
			height: 1040,
		}
	});

	const page = await context.newPage();

	await page.setRequestInterception(true);
	await page.setDefaultNavigationTimeout(90000);

	page.on("request", request => {
		let type = request.resourceType();

		if(type == "image" || type == "font"){
			request.abort();
		} else {
			request.continue();
		}
	});

	await page.goto("https://www.google.com");

	await page.waitFor(5000);

	await page.screenshot({path: "test.png", fullPage: true});

	await browser.close();
})();
```

**Comment by Sue9445** [comment]
Seems solved with v0.11.0
